griffon/services/core_queries.py

```diff
@@ -782,63 +739,43 @@ def execute(self, status=None) -> List[Dict[str, Any]]:
             if self.affect_impact:
                 params["affects__impact"] = self.affect_impact
 
-            res = requests.get(f"{OSIDB_API_URL}/osidb/api/v1/flaws", params=params)
-            flaws = res.json()
-            flaws_cnt = int(flaws["count"])
-            with concurrent.futures.ThreadPoolExecutor() as executor:
-                futures = []
-                flaws = list()
-                for batch in range(0, flaws_cnt, 75):
-                    params["offset"] = batch  # type: ignore
-                    params["limit"] = 75  # type: ignore
-                    futures.append(
-                        executor.submit(
-                            requests.get,
-                            f"{OSIDB_API_URL}/osidb/api/v1/flaws",
-                            params=params,
-                        )
+            # flaws_cnt = self.osidb_session.flaws.count(**params)
+            flaws = self.osidb_session.flaws.retrieve_list_iterator_async(**params)
+
+            for flaw in flaws:
+                for affect in flaw.affects:
+                    if self.affectedness:
+                        if self.affectedness != affect.affectedness:
+                            continue
+                    if self.affect_resolution:
+                        if self.affect_resolution != affect.resolution:
+                            continue
+                    if self.affect_impact:
+                        if self.affect_impact != affect.impact:
+                            continue
+                    affects.append(
+                        {
+                            "link_affect": f"{OSIDB_API_URL}/osidb/api/v1/affects/{affect.uuid}",  # noqa
```

Comment:
`osidb/api/v1` part could be also parametrized for the more re-usability (same goes for corgi url)

Reply:
Did this to isolate the **v1** eg. griffon does not do much handling/checking of version currently and I wanted to make this stick out ... going to leave asis for now.